src/client/components/MusicReserve/index.js

```diff
@@ -126,9 +131,44 @@ export const MusicReserve = ({ userid, setError }) => {
     setAllAlbums(albumsToBeSet);
   };
 
+  const [analytics, setAnalytics] = useState(null);
+  const retrieveAnalytics = () => {
+    if (isInstructorOrAdmin) {
+      const ltik = getLtik();
+      axios.get(`/api/medias/musicres/analytics?ltik=${ltik}`).then(res => {
+        setAnalytics(res.data);
+      });
+    }
+  };
+  useEffect(retrieveAnalytics, [isInstructorOrAdmin]);
+
+  const [showingAnalytics, setShowingAnalytics] = useState(false);
+  const showAnalytics = () => {
+    setShowingAnalytics(!showingAnalytics);
+  };
+
+  if (analytics && showingAnalytics) {
+    return (
+      <>
```

Comment:
@maxpluez  This button like it should be a component since it seems like duplicate code